packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,509 @@
+    description:
+      - httplib2, the library used by the uri module only sends authentication
+        information when a webservice responds to an initial request with a 401
+        status. Since some basic auth services do not properly send a 401,
+        logins will fail. This option forces the sending of the Basic
+        authentication header upon initial request.
+    required: false
+    choices: [ "yes", "no" ]
+    default: "no"
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated. This should only be
+        used on personally controlled sites using self-signed certificates.
+    required: false
+    default: 'yes'
+    choices: ['yes', 'no']
+'''
+
+EXAMPLES = '''
+- name: Create a new repo with name 'my_repo'
+  pulp_repo:
+    name: my_repo
+    relative_url: my/repo
+    state: present
+
+- name: Create a repo with a feed and a relative URL
+  pulp_repo:
+    name: my_centos_updates
+    pulp_host: https://pulp.example.com
+    repo_type: rpm
+    feed_url: http://mirror.centos.org/centos/6/updates/x86_64/
+    relative_url: centos/6/updates
+    url_username: admin
+    url_password: admin
+    force_basic_auth: yes
+    state: present
+
+- name: Remove a repo from the pulp server
+  pulp_repo:
+    name: my_old_repo
+    repo_type: rpm
+    state: absent
+'''
+
+RETURN = '''
+repo:
+  description: Name of the repo that the action was performed on.
+  returned: success
+  type: string
+  sample: repo_name
+'''
+
+
+import json
+
+
+class pulp_server(object):
+    """
+    Class to interact with a Pulp server
+    """
+
+    def __init__(self, module, pulp_host, repo_type):
+        self.module = module
+        self.host = pulp_host
+        self.repo_type = repo_type
+        self.repo_cache = dict()
+
+    def check_repo_exists(self, repo_id):
+        try:
+            repo_config = self.get_repo_config_by_id(repo_id)
+        except IndexError:
+            return False
+        else:
+            return True
+
+    def compare_repo_distributor_config(self, repo_id, **kwargs):
+        repo_config = self.get_repo_config_by_id(repo_id)
+        for distributor in repo_config["distributors"]:
+            for key, value in kwargs.iteritems():
+                if not distributor["config"][key] == value:
+                    return False
+        return True
+
+    def compare_repo_importer_config(self, repo_id, **kwargs):
+        repo_config = self.get_repo_config_by_id(repo_id)
+        for importer in repo_config["importers"]:
+            for key, value in kwargs.iteritems():
+                if value is not None:
+                    if key not in importer["config"].keys():
+                        return False
+                    if not importer["config"][key] == value:
+                        return False
+        return True
+
+    def create_repo(self, repo_id, relative_url, feed_url=None, serve_http=False, serve_https=True, proxy_host=None, proxy_port=None):
```

Comment:
I usually lint as I go in my editor with pep8/flake8. I wrapped this definition because it was starting to get pretty long, but I left many > 80 characters as is recommended for readability in the docs: http://docs.ansible.com/ansible/developing_modules.html#module-checklist